src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ChannelService.java

```diff
@@ -240,4 +248,131 @@ public void unarchiveChannel(Long channelId) {
+    }
+
+    /**
+     * Creates a channel for a course exercise and sets the channel name of the exercise accordingly
+     *
+     * @param exercise    the exercise to create the channel for
+     * @param channelName the name of the channel
+     * @return the created channel
+     */
+    public Channel createExerciseChannel(Exercise exercise, @NotNull String channelName) {
+        if (!exercise.isCourseExercise()) {
+            return null;
+        }
+
+        Channel channelToCreate = new Channel();
+        channelToCreate.setName(channelName);
+        channelToCreate.setIsPublic(true);
+        channelToCreate.setIsAnnouncementChannel(false);
+        channelToCreate.setIsArchived(false);
+        channelToCreate.setDescription(String.format("Channel for %s exercise: %s", exercise.getExerciseType().getExerciseTypeAsReadableString(), exercise.getTitle()));
+        channelToCreate.setExercise(exercise);
+        Channel createdChannel = createChannel(exercise.getCourseViaExerciseGroupOrCourseMember(), channelToCreate, Optional.of(userRepository.getUserWithGroupsAndAuthorities()));
+        exercise.setChannelName(createdChannel.getName());
+        return createdChannel;
+    }
+
+    /**
+     * Create a channel for an exam
+     *
+     * @param exam        the exam to create the channel for
+     * @param channelName the name of the channel
+     * @return the created channel
+     */
+    public Channel createExamChannel(Exam exam, @NotNull String channelName) {
+        if (exam.isTestExam()) {
+            return null;
+        }
+        Channel channelToCreate = new Channel();
+        channelToCreate.setName(channelName);
+        channelToCreate.setIsPublic(false);
+        channelToCreate.setIsAnnouncementChannel(false);
+        channelToCreate.setIsArchived(false);
+        channelToCreate.setDescription("Channel for exam: " + exam.getTitle());
+        channelToCreate.setExam(exam);
+        return createChannel(exam.getCourse(), channelToCreate, Optional.of(userRepository.getUserWithGroupsAndAuthorities()));
+    }
+
+    /**
+     * Update the channel of a lecture
+     *
+     * @param originalLecture the original lecture
+     * @param channelName     the new channel name
+     * @return the updated channel
+     */
+    public Channel updateLectureChannel(Lecture originalLecture, String channelName) {
+        if (channelName == null) {
+            return null;
+        }
+        Channel channel = channelRepository.findChannelByLectureId(originalLecture.getId());
+        return updateChannelName(channel, channelName);
+    }
+
+    /**
+     * Update the channel of an exercise
+     *
+     * @param originalExercise the original exercise
+     * @param updatedExercise  the updated exercise
+     * @return the updated channel
+     */
+    public Channel updateExerciseChannel(Exercise originalExercise, Exercise updatedExercise) {
```

Comment:
I think we should also update the description if the exercise / exam / lecture name gets changed to maintain consistency.

Reply:
We decided to remove the default description for now completely, because setting the description is an instructor's decision anyway.